docs/arch/adr-010-new-versioning-schema.md

```diff
@@ -0,0 +1,145 @@
+
+Consequently, we can initiate the design of our new kURL installation YAML in the following manner:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27”
+```
+
+With the Channel (Kubernetes) now established as the basis for our delivery model, we can enable users to select a Version within this channel, which will allow them to access a specific set of Kubernetes-related features and functionalities (add-ons) that have been thoroughly tested and vetted by our team. This Version serves to abstract each individual add-on version into our own standardised version, ensuring that users can reliably and confidently utilise our offering without worrying about compatibility or interoperability issues.
+
+We will establish the practice of starting each Channel with a version v1.0.0 that groups together the initial set of add-ons (and respective versions) we support for that particular version. As a result, we will need to update the YAML file to reflect this approach, an initial Installer would look like this:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+  version:  "1.0.0"
+```
+While we will delve into the specifics of what the new kURL Version inside a Channel entails in more detail shortly, for the sake of simplicity, bear in mind that this Version essentially groups together a list of add-ons that are compatible with a given Kubernetes version. As such, users can easily upgrade to a newer version within the same Channel by updating the YAML version field and running the installer accordingly. For instance, the following example illustrates how to update the cluster to kURL version v1.8.0 within 1.27 Channel:
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  channel:  "1.27"
+  version:  "1.8.0"
+```
+As we've designed it, users are free to skip as many versions as they wish within a given Channel without any restrictions. This means that they can easily move between kURL versions within the same Channel at their own pace, depending on their individual needs and preferences, without any limitations or requirements.
+
+_From now on this ADR may refer to Channel and Version in its short version "vChannel+kURL Version", for example v1.26+1.1.0 is the equivalent of Channel 1.26 and kURL version 1.1.0_
+
+### About kURL version inside a Channel
+As we've discussed earlier, a kURL Version within a given Channel comprises a list of add-ons, each with their own specific version. With that in mind, we could begin by outlining the initial group of add-ons that would make up kURL Version 1.0.0 within a given Channel 1.26. This version could includes the following set of add-ons, each with their corresponding version (this set of add-ons mentioned is used solely as an example and is not the definitive list of add-ons):
+
+| Add-on | Version |
```

Comment:
Does this also pin a kurl version? How can a user pin the kurl version?

Reply:
im only thinking about reproducibility here for the end user. we have to ensure both the homogeneity of the versions as well as reproducibility by pinning the binary or scripts version used to install k8s and add-ons.